Fix TestChapterRepository_SaveUpsert — nil pointer en CI intégration

## Contexte

Le test `TestChapterRepository_SaveUpsert` (ligne 119 de `chapter_repository_test.go`) panic avec nil pointer dereference dans le workflow CI intégration.

```
panic: runtime error: invalid memory address or nil pointer dereference
chapter_repository_test.go:119
```

Le test fait `repo.Save(ctx, ch)` puis `repo.FindByID(ctx, ch.ID)` — le `FindByID` retourne nil, le test accède à `got.Name` → panic.

## Cause probable

Le `Save` (upsert) échoue silencieusement en CI. Possible que le schema de test n'a pas toutes les colonnes attendues par l'upsert, ou que le `current_revision_id` pose problème.

## Action

- [ ] Ajouter un check d'erreur sur `repo.Save` (ne pas ignorer `_`)
- [ ] Investiguer pourquoi l'upsert échoue en CI mais pas en local
- [ ] Fix le test ou la migration

## Réf

- Run CI : https://github.com/popul/ReviseMieux/actions/runs/23951316354
- `backend/internal/infra/postgres/chapter_repository_test.go:119`

--- a/backend/internal/infra/postgres/chapter_repository_test.go
+++ b/backend/internal/infra/postgres/chapter_repository_test.go
@@ -75,8 +75,12 @@ func TestChapterRepository_FindByUser(t *testing.T) {
 		Subject: "PC", ClassLevel: "4e", Name: "Chapitre 2",
 		Archived: true, CreatedAt: now, UpdatedAt: now,
 	}
-	repo.Save(ctx, ch1)
-	repo.Save(ctx, ch2)
+	if err := repo.Save(ctx, ch1); err != nil {
+		t.Fatalf("Save ch1: %v", err)
+	}
+	if err := repo.Save(ctx, ch2); err != nil {
+		t.Fatalf("Save ch2: %v", err)
+	}
 
 	// Exclude archived
 	active, err := repo.FindByUser(ctx, userID, false)
@@ -109,13 +113,20 @@ func TestChapterRepository_SaveUpsert(t *testing.T) {
 		Subject: "SVT", ClassLevel: "5e", Name: "Original",
 		CreatedAt: now, UpdatedAt: now,
 	}
-	repo.Save(ctx, ch)
+	if err := repo.Save(ctx, ch); err != nil {
+		t.Fatalf("Save (insert): %v", err)
+	}
 
 	ch.Name = "Updated"
 	ch.UpdatedAt = now.Add(time.Hour)
-	repo.Save(ctx, ch)
+	if err := repo.Save(ctx, ch); err != nil {
+		t.Fatalf("Save (upsert): %v", err)
+	}
 
-	got, _ := repo.FindByID(ctx, ch.ID)
+	got, err := repo.FindByID(ctx, ch.ID)
+	if err != nil {
+		t.Fatalf("FindByID: %v", err)
+	}
 	if got.Name != "Updated" {
 		t.Errorf("Name = %q, want Updated", got.Name)
 	}
@@ -149,9 +160,14 @@ func TestChapterRepository_RevisionCRUD(t *testing.T) {
 
 	// Update status
 	rev.Status = chapter.RevisionReady
-	repo.SaveRevision(ctx, rev)
+	if err := repo.SaveRevision(ctx, rev); err != nil {
+		t.Fatalf("SaveRevision (update): %v", err)
+	}
 
-	got, _ = repo.FindRevisionByID(ctx, rev.ID)
+	got, err = repo.FindRevisionByID(ctx, rev.ID)
+	if err != nil {
+		t.Fatalf("FindRevisionByID (after update): %v", err)
+	}
 	if got.Status != chapter.RevisionReady {
 		t.Errorf("Status = %v, want READY", got.Status)
 	}
@@ -377,9 +393,17 @@ func TestChapterRepository_NotionCRUD(t *testing.T) {
 
 	// Upsert
 	n.Name = "Masse volumique (updated)"
-	repo.SaveNotion(ctx, n)
+	if err := repo.SaveNotion(ctx, n); err != nil {
+		t.Fatalf("SaveNotion (upsert): %v", err)
+	}
 
-	notions, _ = repo.FindNotionsByChapter(ctx, ch.ID)
+	notions, err = repo.FindNotionsByChapter(ctx, ch.ID)
+	if err != nil {
+		t.Fatalf("FindNotionsByChapter (after upsert): %v", err)
+	}
+	if len(notions) == 0 {
+		t.Fatal("no notions found after upsert")
+	}
 	if notions[0].Name != "Masse volumique (updated)" {
 		t.Errorf("upserted Name = %q", notions[0].Name)
 	}

